### ISSUE TITLE
zones domain 500 internal server error 

### ISSUE THREAD

**Issue by mandevx** [PowerDNS/pdns#7924]
I am using Server: PowerDNS/4.1.8 I have a strange issue where api return all the zones fine 
curl -v -H 'X-API-Key: mypass' http://127.0.0.1:8081/api/v1/servers/localhost/zones | jq .
but when try to check a particular domain 

curl -v -H 'X-API-Key: mypass' http://127.0.0.1:8081/api/v1/servers/localhost/zones/somedomain.dom | jq .

It returns 500 error 

HTTP/1.1 500 Internal Server Error
< Connection: close
< Content-Length: 21
< Content-Type: text/plain; charset=utf-8
< Server: PowerDNS/4.1.8

But with same api if i enter wrong domain that does not exit return correctly the reply saying as follow : 

{
  "error": "Could not find domain 'balbal.comw.'"
}

Now when I check using "pdnsutil list-zone somedomain.dom"  it just works fine and returns correct record data also when I use Poweradmin ( web admin ) that also shows me every thing correctly.

How can I debug further ?

**Comment by nicovs** [comment]
I have it the other way around, running pdns-server 4.2.0~rc1

List all zones:
`curl -X GET http://localhost:8081/api/v1/servers/localhost/zones -H 'X-API-Key: verySecret' 
Internal Server Error%`

=> PowerDNS logging:
Jun 14 11:34:11 ns1 pdns_server[22958]: HTTP ISE for "/api/v1/servers/localhost/zones": Exception: Unable to convert presentation address 'NULL'


Requesting a single zone works.
Requesting an unexisting zone works too: "Not Found"

**Comment by pieterlexis** [comment]
Please provide the content of the database's domains and records table or a minimal, reproducible example so we can debug this.

**Comment by nicovs** [comment]
Hi Pieter,

Ty for your answer! 
While i was debugging furthers, i noticed RC2 was released.
With that release, a better debug output was provider:

> Error: stoul; (invalid argument during std::stoul); data was "d-cra039a.eu"

Which pointed me to a zone which had a space in it... Probably some legacy stuff still in the databases from IDN converts domain named.
As soon as I removed this zone: all working as intended!

Tnx!
Nico
